lib/bookingsync/engine/account_model.rb

```diff
@@ -0,0 +1,114 @@
+class BookingsyncEngineUniquenessValidator < ActiveRecord::Validations::UniquenessValidator
+  def initialize(options)
+    if options[:scope].present? && options[:scope].respond_to?(:call)
+      options[:scope] = options[:scope].call
+    end
+    super
+  end
+end
+
+module BookingSync::Engine::AccountModel
+  extend ActiveSupport::Concern
+
+  included do
+    validates :synced_id, bookingsync_engine_uniqueness: {
+      scope: Proc.new { BookingSync::Engine.support_multi_applications? ? :host : nil }
```

Comment:
do we really need a Proc here instead of a lambda? I think the result will be the same. It's best to use lambdas over procs as procs have pretty unique way of dealing with return values.